# Human Metapneumovirus - All Clades with Reference Genome NC_039199

| Key                    | Value                                                                                                               |
| ---------------------- | --------------------------------------------------------------------------------------------------------------------|
| authors                | Isabel Joia, [Richard Neher](https://neherlab.org), [Nextstrain](https://nextstrain.org)                         |
| data source            | Genbank                                                                                |
| workflow               | [github.com/nextstrain/human-metapneumovirus](https://github.com/nextstrain/human-metapneumovirus)                                                      |
| nextclade dataset path | nextstrain/hmpv/all-clades/NC_039199                                                                                     |
| reference              | NC_039199                                                                                                      |
| clade definitions      | [github.com/nextstrain/human-metapneumovirus/nextclade/clades.tsv](https://github.com/nextstrain/human-metapneumovirus/blob/main/nextclade/resources/clades.tsv)              |


## Scope of this dataset
This dataset for human metapneumovirus uses reference sequence [NC_039199](https://www.ncbi.nlm.nih.gov/nuccore/NC_039199.1/) which is available at NCBI RefSeq. The reference tree covers the diversity of human metapneumovirus since the first sequenced samples.
The reference tree is mid-point rooted.

**Note: The reference sequence belongs to clade A1 and extensive divergence between clades A and B may render alignment in gene G unreliable for sequences in clade B.**

### Features
This dataset supports:
 - Assignment to lineage nomenclature initially proposed by [van den Hoogen et al, 2004](https://wwwnc.cdc.gov/eid/article/10/4/03-0393_article)
 - Phylogenetic placement
 - Sequence QC (though frame shifts in the G-protein can not be accurately called across all of HMPV diversity)


